provider/spark.go

```diff
@@ -1697,115 +1700,66 @@ func (spark *SparkOfflineStore) updateQuery(query string, mapping []SourceMappin
 	return updatedQuery, sources, nil
 }
 
-// TODO: delete this method in favor of Filepath methods
-func (spark *SparkOfflineStore) getSourcePath(path string) (string, error) {
-	fileType, fileName, fileVariant := spark.getResourceInformationFromFilePath(path)
-	var filePath string
-	if fileType == "primary" {
-		fileResourceId := ResourceID{Name: fileName, Variant: fileVariant, Type: Primary}
-		fileTable, err := spark.GetPrimaryTable(fileResourceId)
+func (spark *SparkOfflineStore) getSourcePath(tableName string) (string, error) {
+	logger := spark.Logger.With("table", tableName)
+	resourceType, name, variant, err := ps.TableNameToResource(tableName)
```

Comment:
Isn't there a struct we could parse into? If not consider making one

Reply:
I'd love to parse this into `ResourceID` but I can include it in `provider_schema` without running into a circular dependency. I would like to move `ResourceID` into `provider_schema` eventually.